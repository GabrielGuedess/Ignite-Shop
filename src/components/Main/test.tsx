import { render, screen } from '@testing-library/react';

import { Main } from '.';

describe('<Main />', () => {
  it('should render the heading correctly', () => {
    const { container } = render(
      <Main
        title="React Avançado"
        description="TypeScript, ReactJS, NextJS e Styled Components"
      />,
    );

    expect(
      screen.getByRole('heading', { name: /react avançado/i }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
