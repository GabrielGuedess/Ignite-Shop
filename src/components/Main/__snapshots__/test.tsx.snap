// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading correctly 1`] = `
<main
  class="c-oFMmS"
>
  <img
    alt="Imagem de um átomo e React Avançado escrito ao lado."
    class="c-iyMmUv"
    src="/img/logo.svg"
  />
  <h1
    class="c-iKvxSq"
  >
    React Avançado
  </h1>
  <h2
    class="c-hipMmg"
  >
    TypeScript, ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Um desenvolvedor de frente para uma tela com código."
    class="c-dHOelJ"
    src="/img/hero-illustration.svg"
  />
</main>
`;
